components/inputs/input-number.js

```diff
@@ -17,6 +17,12 @@ const HINT_TYPES = {
 	INTEGER: 4
 };
 
+// US137000 - prevent Lit default converter from converting undefined to 0
+const numberConverter = {
```

Comment:
Reasoning behind these changes, for future reference: 

This has to do with the changes in how attributes are converted to properties in Lit 2.
In Lit 1: https://lit.dev/docs/v1/components/properties/#conversion For Numbers, when the attribute is defined, set the property to `Number(attributeValue)`.

In Lit 2: https://lit.dev/docs/components/properties/#conversion-type If the element has the corresponding attribute, set the property to `Number(attributeValue)`.

In Lit 1 passing in `undefined` does not affect the property (since it's undefined).
In Lit 2, the undefined gets automatically converted to 0.

To emulate the Lit 1 behaviour for compatibility reasons, we need to create this custom converter that converts null/undefined to undefined, and parses numbers as expected.